packages/aft/lib/src/commands/generate/generate_workflows_command.dart

```diff
@@ -127,7 +145,64 @@ jobs:
           );
         }
       }
+
       workflowFile.writeAsStringSync(workflowContents.toString());
+
+      // Add workflow for Android unit tests if needed
+      if (!needsAndroidTest) {
+        continue;
+      }
+      final androidWorkflowFilepath = p.join(
+        rootDir.path,
+        '.github',
+        'workflows',
+        '${package.name}_android.yaml',
+      );
+
+      final androidExternalPackagePaths = [
+        if (externalAndroidPackageTestDirExists)
+          '${repoRelativePath}_android/**/*'
+      ];
+      final androidWorkflowPaths = [
+        '.github/workflows/$androidWorkflow',
+        p.relative(androidWorkflowFilepath, from: rootDir.path)
+      ];
+      final androidPathString =
+          (androidExternalPackagePaths + androidWorkflowPaths)
+              .map((path) => "      - '$path'")
+              .join('\n');
+
+      final androidWorkflowFile = File(androidWorkflowFilepath);
+      final androidWorkflowContents = '''
+# Generated with aft. To update, run: `aft generate workflows`
+name: ${package.name} Android
+on:
+  push:
+    branches:
+      - main
+      - stable
+      - next
+  pull_request:
+    paths:
+      - '$repoRelativePath/**/*.yaml'
+      - '$repoRelativePath/android/**/*'
+      - '$repoRelativePath/example/android/**/*'
+$androidPathString
+  schedule:
+    - cron: "0 0 * * 0" # Every Sunday at 00:00
+defaults:
+  run:
+    shell: bash
+permissions: read-all
+
+jobs:
+  test:
+    uses: ./.github/workflows/$androidWorkflow
+    with:
+      working-directory: $repoRelativePath
+''';
+
+      androidWorkflowFile.writeAsStringSync(androidWorkflowContents);
```

Comment:
Can we move all the new logic to its own method since it seems completely independent of the rest?

Reply:
good suggestion, done. This will also help once iOS added here.